common/lib/xmodule/xmodule/partitions/partitions.py

```diff
@@ -40,7 +40,7 @@ class Group(namedtuple("Group", "id name")):
 
     def __new__(cls, id, name):
         # pylint: disable=super-on-old-class
-        return super(Group, cls).__new__(cls, int(id), name)
+        return super(Group, cls).__new__(cls, id, name)
```

Comment:
My main concern with this is what would happen if we rolled back the code after string ID values had been written.  I can imagine some very strange and difficult-to-debug conditions, since strings and numbers are not equivalent in Python.

For the `Group` id, I think we can still use numbers.  We'd just set the constants `NON_VERIFIED = 0`, `VERIFY_ALLOW = 1`, and `VERIFY_DENY = 2`.

For the `UserPartition` we were planning to use the ID to encode the namespace ("verification") and location.  We needed this to be able to (a) find verification partitions, and (b) figure out which partitions are no longer used so we can delete them.

It seems to me that if we used integer `id` numbers, we could still do (a) and (b).  We could do (a) because verification partitions will all have the scheme set to our verification scheme.  And we can do (b) because we are encoding the location as a `parameter` in the user partition.

@zubair-arbi I know this is a change from the design document, but would it be possible to use integer `id` values for both `Group` and `UserPartition`?  Is there an aspect of this problem that I'm not understanding?